lib/import/brca/providers/salisbury/salisbury_handler.rb

```diff
@@ -13,56 +13,223 @@ def process_fields(record)
             genotype.add_passthrough_fields(record.mapped_fields,
                                             record.raw_fields,
                                             PASS_THROUGH_FIELDS)
-
-            process_molecular_testing(genotype, record)
             add_organisationcode_testresult(genotype)
-            extract_teststatus(genotype, record)
             add_provider_code(genotype, record, ORG_CODE_MAP)
-            results = process_variant_record(genotype, record)
-            results.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
+            record.raw_fields.reject! do |raw_record|
+              raw_record['status'].match(/Variant\sfor\sAlissa\sreview/ix)
+            end
+
+            # For clarity, `raw_fields` contains multiple raw records for same SRI
+            assign_molecular_testing_var(record)
+            res = process_record(genotype, record)
+            res.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
+          end
+
+          def assign_molecular_testing_var(record)
+            @mol_testing_type = record.raw_fields.pluck('moleculartestingtype')&.uniq&.first&.downcase
+          end
+
+          def add_provider_code(genotype, record, org_code_map)
+            raw_org = record.raw_fields.pluck('providercode')&.uniq&.first&.downcase
+            org_code = org_code_map[raw_org]
+            return if org_code.blank?
+
+            genotype.attribute_map['providercode'] = org_code
           end
 
-          def process_molecular_testing(genotype, record)
-            mol_testing_type = record.raw_fields['moleculartestingtype']&.downcase
-            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAPPING[mol_testing_type])
-            scope = TEST_SCOPE_MAPPING[mol_testing_type].presence || :no_genetictestscope
+          def process_record(genotype, record)
+            process_molecular_testing(genotype)
+            genotypes = []
+            if ROW_LEVEL.include?(@mol_testing_type)
+              record.raw_fields.each { |raw_record| process_row_case(genotypes, genotype, raw_record) }
+            elsif PANEL_LEVEL.keys.include?(@mol_testing_type)
+              process_panel_case(genotypes, genotype, record)
+            elsif HYBRID_LEVEL.include?(@mol_testing_type)
+              process_hybrid_case(genotypes, genotype, record)
+            end
+
+            genotypes
+          end
+
+          def process_row_case(genotypes, genotype, record)
+            genotype_new = genotype.dup
+            assign_status_var(record)
+            status = extract_teststatus_record
+            genotype_new.add_status(status)
+            extract_gene_row(genotype_new, record)
+            if [2, 10].include? status
+              assign_variantpathclass_record(genotype_new)
+              variant = record['genotype']
+              process_variants(genotype_new, variant) if positive_record?(genotype_new) && variant.present?
```

Comment:
Do we need to use 'is positive_record' here? Does that mean we are missing out on capturing variants with test status of 10?

Reply:
If you look at "positive record?" it includes those with test status 10 - so maybe the name just needs to be changed

Reply:
Ah I accidentally looked at the method removed instead of the new one. Same comment as Laura's then :-)

Reply:
Removed the method in latest commit.